### PR TITLE
Converting iterator component to Lit

### PR DESCRIPTION
**Pull request by dlockhart**
I wasn't planning on doing this so soon, but this was the only component that was hooked up to Serge in a rather roundabout way that required Gulp to generate a bunch of files. So I decided to convert this first, and then we'll have the right localization stuff in place for future Lit things.

This also unlocked being able to rename stuff back to `.js` from `.mjs`, which is why there's a lot of files affected.

### CODE REVIEW

File: components/build/lang/fi.js [hunk 1]
```diff
@@ -1,17 +0,0 @@
...
-
-import { dedupingMixin } from '@polymer/polymer/lib/utils/mixin.js';
-
-/* @polymerMixin */
-const LangFiImpl = (superClass) => class extends superClass {
-	constructor() {
-		super();
-		this.fi = {
-			'next': 'Next',
-			'previous': 'Previous'
```
Comment (dlockhart) [inline components/build/lang/fi.js:11]: I just deleted this one... Finnish isn't something we support and it's English anyway.

File: d2l-navigation-button.js [hunk 2]
```diff
@@ -16,6 +16,11 @@ class D2LNavigationButton extends mixinBehaviors([D2L.PolymerBehaviors.Button.Be
 
 	static get properties() {
 		return {
+			disabled: {
+				type: Boolean,
+				refelctToAttribute: true,
+				value: false
+			},
```
Comment (dlockhart) [inline d2l-navigation-button.js:23]: The Lit analyzer found this missing property because iterator (now Lit) was setting it.

File: package.json [hunk 3]
```diff
@@ -1,21 +1,21 @@
...
   "name": "@brightspace-ui-labs/navigation",
   "version": "5.3.2",
+  "type": "module",
   "repository": "https://github.com/BrightspaceUILabs/navigation.git",
   "scripts": {
-    "build": "gulp build",
     "lint": "npm run lint:eslint && npm run lint:lit && npm run lint:style",
     "lint:eslint": "eslint . --ext .js,.html",
-    "lint:lit": "lit-analyzer d2l-navigation.js d2l-navigation-separator.js --strict --rules.no-unknown-tag-name off",
+    "lint:lit": "lit-analyzer d2l-navigation.js d2l-navigation-separator.js ./components/d2l-navigation-iterator/d2l-navigation-iterator-item.js --strict --rules.no-unknown-tag-name off",
```
Comment (dlockhart) [inline package.json:9]: Once everything is Lit, this should be easy to clean up.

File: test/iterator.visual-diff.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
 /* eslint-disable no-invalid-this */
 /* global forceFocusVisible */
-import puppeteer from 'puppeteer';
 import { oneEvent, VisualDiff } from '@brightspace-ui/visual-diff';
+import puppeteer from 'puppeteer';
```
Comment (dlockhart) [inline test/iterator.visual-diff.js:4]: Lint fix.

File: components/d2l-navigation-iterator/d2l-navigation-iterator-item.js [hunk 5]
```diff
@@ -1,132 +1,117 @@
...
 
-/**
- * @customElement
- * @polymer
- */
-class D2LNavigationIteratorItem extends NavigationLocalize(PolymerElement) {
-	static get is() { return 'd2l-navigation-iterator-item'; }
+class NavigationIteratorItem extends LocalizeNavigationElement(LitElement) {
 
 	static get properties() {
```
Comment (svanherk) [inline components/d2l-navigation-iterator/d2l-navigation-iterator-item.js:10]: Minor: these could be condensed to one line each

**Review by svanherk — APPROVED** [review]

File: components/d2l-navigation-iterator/d2l-navigation-iterator-item.js [hunk 6]
```diff
@@ -1,132 +1,117 @@
...
 	}
 
-	_typeChanged(newValue, oldValue) {
-		const validTypes = ['previous', 'next'];
-		if (validTypes.indexOf(newValue) === -1) {
-			if (validTypes.indexOf(oldValue) === -1) {
-				this.type = 'previous';
-			} else {
-				this.type = oldValue;
-			}
```
[the remark this answers is not in the document]
Comment (dlockhart) [reply components/d2l-navigation-iterator/d2l-navigation-iterator-item.js:126]: Haha seemed like overkill. It's still going to default into a "previous" state if they give it some weird value.

**Pull request merged**